src/layers/repository/device_repo.py

```diff
@@ -0,0 +1,255 @@
+from domain.core.device import (
+    Device,
+    DeviceCreatedEvent,
+    DeviceDeletedEvent,
+    DeviceKey,
+    DeviceKeyAddedEvent,
+    DeviceKeyRemovedEvent,
+    DevicePage,
+    DevicePageAddedEvent,
+    DevicePageRemovedEvent,
+    DevicePageUpdatedEvent,
+    DeviceUpdatedEvent,
+    Relationship,
+    RelationshipAddedEvent,
+    RelationshipRemovedEvent,
+)
+from repository.keys import (
+    device_key_sk,
+    device_page_sk,
+    device_pk,
+    device_relationship_sk,
+    ods_pk,
+)
+
+from .errors import NotFoundException
+from .repository import Repository
+from .utils import marshall, unmarshall
+
+
+class DeviceRepository(Repository[Device]):
+    def __init__(self, table_name, dynamodb_client):
+        super().__init__(
+            table_name=table_name, model=Device, dynamodb_client=dynamodb_client
+        )
+
+    def handle_DeviceCreatedEvent(self, event: DeviceCreatedEvent, entity: Device):
+        return {
+            "Put": {
+                "TableName": self.table_name,
+                "Item": marshall(
+                    {
+                        "pk": device_pk(event.id),
+                        "sk": device_pk(event.id),
+                        "pk_1": ods_pk(event.ods_code),
+                        "sk_1": device_pk(event.id),
+                        "name": event.name,
+                        "type": event.type,
+                        "ods_code": event.ods_code,
+                        "product_team_id": event.product_team_id,
+                        "status": event.status,
```

Comment:
after switching to dataclasses you can do `**asdict(event)` here to avoid data integrity issues when we extend or change models (missing data)

Reply:
ok, pick this up in your ticket.